sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/XmlFile.java

```diff
@@ -95,10 +97,11 @@ private void checkForCharactersBeforeProlog(FileSystem fileSystem) {
     }
   }
 
-  private void checkBOM(String line) {
-    if (line.startsWith(BOM_CHAR)) {
-      characterDeltaForHighlight = -1;
+  private static boolean isFileBeginning(int line, int column, boolean hasBOM) {
```

Comment:
@pynicolas Are you sure about this method name? As I understand, it does the opposite